pkg/daemon/ceph/osd/kms/vault_api.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kms
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/libopenstorage/secrets/vault"
+	"github.com/pkg/errors"
+
+	"github.com/hashicorp/vault/api"
+)
+
+const (
+	kvVersionKey = "version"
+	kvVersion1   = "kv"
+	kvVersion2   = "kv-v2"
+)
+
+// NewVaultClient returns a vault client, there is no need for any secretConfig validation
+// Since this is called after an already validated call InitVault()
+func NewVaultClient(secretConfig map[string]string) (*api.Client, error) {
+	// DefaultConfig uses the environment variables if present.
+	config := api.DefaultConfig()
+
+	// Configure TLS
+	if err := configureTLS(config, secretConfig); err != nil {
+		return nil, err
+	}
+
+	// Initialize the vault client
+	client, err := api.NewClient(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func IsKvBackendV2(client *api.Client, secretConfig map[string]string) (bool, error) {
+	backendPath := GetParam(secretConfig, vault.VaultBackendPathKey)
+	if backendPath == "" {
+		backendPath = vault.DefaultBackendPath
+	}
+
+	backend := GetParam(secretConfig, vault.VaultBackendKey)
```

Comment:
Why do we need to still support this if auto-detection works? This won't have a backward compatibility issue right?

Reply:
Whoever passes a parameter expects it to be interpreted even if it happens to be incorrect. This could be useful for checking errors too.